packages/server/src/index.ts

```diff
@@ -0,0 +1,204 @@
+import cors from 'cors';
+import { ethers, BytesLike } from 'ethers';
+import { Fragment, FunctionFragment, Interface, JsonFragment } from '@ethersproject/abi';
+import { hexlify } from '@ethersproject/bytes';
+import express from 'express';
+
+interface RPCCall {
+  id: string;
+  data: {
+    to: BytesLike;
+    data: BytesLike;
+  };
+}
+
+interface RPCResponseBase {
+  jobRunID: string;
+  statusCode: number;
+}
+
+interface RPCSuccessResponse extends RPCResponseBase {
+  statusCode: 200;
+  data: {
+    result: BytesLike;
+  };
+}
+
+interface RPCClientErrorResponse extends RPCResponseBase {
+  statusCode: 400 | 404;
+  error: {
+    name: string;
+    message: string;
+  };
+}
+
+interface RPCServerErrorResponse extends RPCResponseBase {
+  statusCode: 400;
+  error: {
+    name: string;
+    message: string;
+  };
+}
+
+type RPCResponse = RPCSuccessResponse | RPCClientErrorResponse | RPCServerErrorResponse;
+
+export type HandlerFunc = (args: ethers.utils.Result, req?: RPCCall) => Promise<Array<any>> | Array<any>;
+
+interface Handler {
+  type: FunctionFragment;
+  func: HandlerFunc;
+}
+
+function toInterface(abi: string | readonly (string | Fragment | JsonFragment)[] | Interface) {
+  if (Interface.isInterface(abi)) {
+    return abi;
+  }
+  return new Interface(abi);
+}
+
+function isRPCCall(x: any): x is RPCCall {
+  return x.id !== undefined && x.data !== undefined && x.data.to !== undefined && x.data.data !== undefined;
+}
+
+export interface HandlerDescription {
+  type: string;
+  func: HandlerFunc;
+}
+
+/**
+ * Implements a Durin gateway service using express.js.
+ *
+ * Example usage:
+ * ```javascript
+ * const ccipread = require('ccip-read');
```

Comment:
npm module name is `@smartcontractkit/ccip-read-server` but you load as `ccip-read`? Looks a bit confusing